### PR TITLE
kubelet/rkt: minor updates to match rkt v0.7.0.

### PR DESCRIPTION
**Pull request by yifan-gu**
This is mainly to keep up with the rkt cli, which now supports multiple apps with the same image in a pod, thus here we need to change the reference to the app from `imageid` to `appname`.

Besides, this is a split from #8951 Next follow up PR should be adding `PullImage` and related fixes. Thanks :)

/cc @dchen1107 @yujuhong

### CODE REVIEW

File: pkg/kubelet/rkt/pod_info.go [hunk 1]
```diff
@@ -38,6 +38,8 @@ const (
 	Deleting       = "deleting" // This covers pod.isExitedDeleting and pod.isDeleting.
 	Exited         = "exited"   // This covers pod.isExited and pod.isExitedGarbage.
 	Garbage        = "garbage"
+
+	exitCodePrefix = "app-"
```
Comment (dchen1107) [inline pkg/kubelet/rkt/pod_info.go]: This variable name is really confusion here. Rename it to appcPrefix?

File: pkg/kubelet/rkt/pod_info.go [hunk 2]
```diff
@@ -49,9 +51,9 @@ type podInfo struct {
...
 	// The pid of the init process in the pod.
 	pid int
-	// A map from image hashes to exit codes.
-	// TODO(yifan): Should be appName to exit code in the future.
+	// A map of [app name]:[exit code].
 	exitCodes map[string]int
+	// TODO(yifan): Expose [app name]:[image id].
 }
 
 // parsePodInfo parses the result of 'rkt status' into podInfo.
```
Comment (dchen1107) [inline pkg/kubelet/rkt/pod_info.go]: Can we have a sample output of rkt status here as a comment? I remembered that we used to have that piece of information here.

Comment (yifan-gu) [inline pkg/kubelet/rkt/pod_info.go]: Good catch! Thanks :)

File: pkg/kubelet/rkt/pod_info.go (hunk 1 above)
Comment (yifan-gu) [inline pkg/kubelet/rkt/pod_info.go]: The output for `rkt status $UUID` is now:
```
state=exited
pid=-1
exited=true
app-etcd=0
app-redis=0
```

So `app-` is the prefix before the app name for each app's exit code. (We definitely need some structured data here).

I think I need to add some doc here.

File: pkg/kubelet/rkt/pod_info.go (hunk 2 above)
Comment (dchen1107) [inline pkg/kubelet/rkt/pod_info.go]: the output of rkt status keep changing, and I expects it is going to change in the future until we stabilize it. Without that output in the comment, it is very hard to review. For example the above question I raised, I guessed that is the case you just answered, but not sure about it. :-)

Comment (yifan-gu) [inline pkg/kubelet/rkt/pod_info.go]: @dchen1107 Updated

**Comment by dchen1107** [comment]
LGTM

**Pull request merged**